parameter/DomainConfiguration.cpp

```diff
@@ -397,8 +397,10 @@ void CDomainConfiguration::validate(const CConfigurableElement* pConfigurableEle
 {
     CAreaConfiguration* pAreaConfigurationToValidate = getAreaConfiguration(pConfigurableElement);
 
-    // Delegate
-    pAreaConfigurationToValidate->validate(pMainBlackboard);
+    if (pAreaConfigurationToValidate != NULL) {
```

Comment:
Refer to getAreaConfiguration() function implementation we cannot avoid the NULL return since the function should retrun an address. But i agree with you  about system behaviour in error case.  We can abort in  to getAreaConfiguration() avoid this case.